driver-core/src/test/functional/com/mongodb/client/model/expressions/BooleanExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings({"PointlessBooleanExpression", "ConstantConditions", "ConstantConditionalExpression"})
+class BooleanExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
```

Comment:
We should have unit tests that assert the generated MQL along with the integration tests that assert that the MQL does what we expect it to do.

Reply:
```
assertExpression(true || false, tru.or(fal), "{'$or': [true, false]}");
```

`assertExpression`  asserts that each expression evaluates to the provided result on the left, and that the expression generates the appropriate MQL on the right.

Reply:
Oh, I missed that.  Thanks.